src/Maybe.ts

```diff
@@ -0,0 +1,108 @@
+import { Applicative, Monad, Traversable } from './types';
+
+type nothing = undefined | null;
```

Comment:
No void represent the absence of value it is meant to type function with no return value.

It's true that it is a supertype of null and undefined, but it function strangely.
For example 

```js
function foo(): void {
    return null;
}
```
Is invalid : `Impossible d'assigner le type 'null' au type 'void'.`

Also note that they discourage its use as a type parameter in the link you provided:

`NOTE: We might consider disallowing declaring variables of type Void as they serve no useful purpose. However, because Void is permitted as a type argument to a generic type or function it is not feasible to disallow Void properties or parameters.`